test/functional/schema-builder/change-column.ts

```diff
@@ -22,6 +41,49 @@ describe("schema builder > change column", () => {
     beforeEach(() => reloadTestingDatabases(connections))
     after(() => closeTestingConnections(connections))
 
+    it("emits ALTER, not DROP/ADD when increasing varchar length", async () => {
+        for (const base of connections) {
+            // We choose postgres
+            if (base.options.type !== "postgres") continue
+
+            // 1) initial schema (varchar(50)) + seed
+            const ds1 = new DataSource({
+                ...(base.options as any),
+                entities: [BugInitial],
+                synchronize: true,
+                dropSchema: false, // ensure we keep the table
+            } as any)
```

Comment:
This test is still not written correctly. Please read the other tests in this file and use them as a model for how to write a test for this feature.